osquery/events/events.cpp

```diff
@@ -416,14 +421,18 @@ Status EventFactory::run(EventPublisherID& type_id) {
 
 void EventFactory::end(bool should_end) {
   auto& ef = EventFactory::getInstance();
-  ef.ending_ = should_end;
```

Comment:
Previously, checking an isEnding or `ending_` in the event factory might have produced an error is the main thread had already exited.